### DISCUSSION TITLE
How to preserve selection in a TreeDataGrid?

### DISCUSSION THREAD

**Discussion by DevAM-Tools** [AvaloniaUI/Avalonia#14105] [Q&A]
I have a `TreeDataGrid` that gets frequently updated by adding multiple items every 500ms. On every update a Reset event is triggerd for the underlaying `ObservableCollectionExtended`. Every time the collection is updated the `TreeDataGrid` loses its selection.

How can I preserve the selection when the underlying collection is updated?

**Comment by timunie** [comment]
maybe save a reference of the selected items and add the selection back. But I feel like DynamicDatas SourceCache maybe better here as that has the option to "AddOrUpdate" by ID and thus old items stay in the collection
